src/coreclr/jit/lsra.cpp

```diff
@@ -196,9 +196,9 @@ BasicBlock::weight_t LinearScan::getWeight(RefPosition* refPos)
             if (refPos->getInterval()->isSpilled)
             {
                 // Decrease the weight if the interval has already been spilled.
-                if (varDsc->lvLiveInOutOfHndlr)
+                if (varDsc->lvLiveInOutOfHndlr || refPos->getInterval()->firstRefPosition->singleDefSpill)
                 {
-                    // An EH var is always spilled at defs, and we'll decrease the weight by half,
+                    // An EH-var/single-def is always spilled at defs, and we'll decrease the weight by half,
                     // since only the reload is needed.
                     weight = weight / 2;
                 }
```

Comment:
As we discussed offline the code in the `else` below this is likely not doing what one might expect, especially for PGO. The reduction should be relative to some other weight (method entry, etc). Don't need to fix it now.

Reply:
Added this point in https://github.com/dotnet/runtime/issues/55289#issuecomment-876570901.